cli/container.go

```diff
@@ -20,6 +20,20 @@ import (
 	"github.com/docker/docker/client"
 )
 
+type ClusterSpec struct {
```

Comment:
Maybe at some point we can add a new level here for server and agents, if we add more options?
E.g. you would call the server args via `clusterSpec["server"].Args`
Achieved by a new struct for `ServerSpec` and one for `AgentSpec` and embedding them into the `ClusterSpec` struct.
WDYT?

Reply:
I see your view point and agree that we should consider this approach down the road. 

Right now, both server and agent share a lot of information, the clusterSpec is a simple union of serverSpec and agentSpec. It seems to be a good abstraction.  At the same time, I do not oppose the way you have suggested.